### DISCUSSION TITLE
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'csrf')

### DISCUSSION THREAD

**Discussion by Etzahu** [livewire/livewire#7920] [1. Help]
When I submit in the forms of any livewire 3 component, I get the following message, I can't find the reason why 
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'csrf')

![error](https://github.com/livewire/livewire/assets/36831481/455bc414-693c-411a-a7be-ed8515b487cd)

**Comment by AidasK** [comment]
You probably have disabled csrf generation by removing the session middleware. Make sure sessions are working. Also you should check if your layout/output html contains csrf token too. If it doesn't you should include @csrf tag.
